runtime/kubernetes/container.go

```diff
@@ -172,7 +173,14 @@ func (c *client) SetupContainer(ctx context.Context, ctn *pipeline.Container) er
 		}
 	}
 
-	// TODO: add SecurityContext options (runAsUser, runAsNonRoot, sysctls)
+	if c.PipelinePodTemplate != nil && c.PipelinePodTemplate.Spec.Container != nil {
+		securityContext := c.PipelinePodTemplate.Spec.Container.SecurityContext
+
+		// TODO: add more SecurityContext options (runAsUser, runAsNonRoot, sysctls)
```

Comment:
I did not address these in this PR because:
- fixing the user issues is not straight forward:
    -  `runAsUser` is an int user id, but the `user` tag in the pipeline is a string, and
    - `runAsNonRoot` is a form of validation - it just causes containers to fail if the image is not configured to use root; it does not switch it to use a non-root user.
- `sysctls` requires more discussion and safety checking:
    - it might need some new tag(s) in the pipeline config and
    - this will need some kind of sanity-checking validation, perhaps only allowing a limited set of them. The kubernetes docs warn that only a subset of sysctls are namespaced sufficiently to use safely per-container, but otherwise should be handled at the node-level.

Reply:
For future reference:

Also, I added `runAsNonRoot` for the pod SecurityContext, but I did not add in the container SecurityContext. We may need to add it at container-level as well when this becomes configurable from the pipeline.

`sysctls` is only in the pod-level SecurityContext, and I did implement it there. We may still need to add something to pipeline config and add safety checking around that.

Had I realized that when writing this comment, I would have dropped `sysctls` and possibly `runAsNonRoot` from this TODO.
:shrug: